### PR TITLE
LKE-10021: Update eslint

### PR DESCRIPTION
**Pull request by gmasclet**
https://linkurious.atlassian.net/browse/LKE-10021

Update `eslint`.

### CODE REVIEW

File: src/api/commonTypes.ts [hunk 1]
```diff
@@ -101,8 +100,8 @@ export interface CommentMention {
 }
 
 export interface DeletableUser {
-  username: string | DeletedUser;
-  email: string | DeletedUser;
```
[the remark this answers is not in the document]
Comment (davidrapin) [reply src/api/commonTypes.ts:105]: any risk of triggering warnings downstream? (e.g; frontend)

**Review by davidrapin — APPROVED** [review]
No blocking issues.

When updating libraries used across several repos (like eslint), if you don't do already, I recommend you do it from an umbrella checkout, so that you can test that you are not breaking the "top level" npm install.

**Review by davidrapin — CHANGES_REQUESTED** [review]
package-lock update missing

File: src/api/commonTypes.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (gmasclet) [reply src/api/commonTypes.ts:105]: We've checked with @EricMallard. Unless we're mistaken, there's no risk

**Comment by gmasclet** [comment]
@davidrapin not sure why the package-lock was missing. 

Anyway, I've rebased this PR, as the mocha update has been merged (and this was causing conflicts). You can see the package-lock is now present. 

One side effect of having run `npm i` is that the lockfile version has been upgraded from 2 to 3. Not a big deal in my opinion, we have to do it anyway. I suggest we move forward (and progressively upgrade the lockfile version on the other repositories).

**Comment by davidrapin** [comment]
@gmasclet if you need to force a lockfile version, use `npm i --lockfile-version 2` 
keep in mind that incompatible version of lockfile can require the CI to regenerate the full package-lock, which is kinda slow. We may want to upgrade all lockfile versions at once.

**Review by davidrapin — APPROVED** [review]

**Comment by gmasclet** [comment]
@davidrapin Makes sense, I've reverted the lockfile version to 2 :+1:

**Pull request merged**